### PR TITLE
add relayer whitelist

### PR DESCRIPTION
**Pull request by liu-zhipeng**
## Checklist

- [ ] Test manually on `test-ui` using remote chains.
- [ ] Run load tests.
- [ ] Update documentation if needed.
- [ ] Update CHANGELOG.md.

### CODE REVIEW

File: packages/deployments/contracts/contracts/Connext.sol [hunk 1]
```diff
@@ -273,6 +282,19 @@ contract Connext is
...
+   * @param approved - approved or not
+   */
+  function setupRelayer(address relayer, bool approved) external override onlyOwner {
+    if (relayer == address(0)) revert Connext__setupRelayer_badAddress();
+    if (approvedRelayers[relayer] == approved) revert Connext__setupRelayer_alreadyDid();
+    approvedRelayers[relayer] = approved;
+
+    emit RelayerSetup(relayer, approved);
+  }
+
```
Comment (LayneHaber) [inline packages/deployments/contracts/contracts/Connext.sol:297]: I'm inclined to have separate functions for adding and removing relayers (same as we do with assets/routers)

File: packages/deployments/contracts/contracts/Connext.sol [hunk 2]
```diff
@@ -515,7 +537,7 @@ contract Connext is
       // Send funds to the user
       AssetLogic.transferAssetFromContract(adopted, _args.params.to, amount, wrapper);
     } else {
-      // Send funds to interprepter
+      // Send funds to executor
```
Comment (LayneHaber) [inline packages/deployments/contracts/contracts/Connext.sol:540]: 👍

File: packages/deployments/contracts/contracts/Connext.sol [hunk 3]
```diff
@@ -679,6 +701,11 @@ contract Connext is
       return;
     }
 
+    // If the sender is not approved relayer, revert()
+    if (!approvedRelayers[msg.sender]) {
+      revert Connext__handleRelayerFees_notApprovedRelayer();
+    }
```
Comment (LayneHaber) [inline packages/deployments/contracts/contracts/Connext.sol:707]: @rhlsthrm we need the address of the gelato exec contract to make sure we don't break testnet

File: packages/deployments/contracts/contracts/interfaces/IConnext.sol [hunk 4]
```diff
@@ -131,6 +131,13 @@ interface IConnext {
    */
   event AssetRemoved(bytes32 canonicalId, address caller);
 
+  /**
+   * @notice Emitted when a rlayer is added or removed from whitelists
+   * @param relayer - The relayer address to be added or removed
+   * @param approved - Added or removed?
+   */
+  event RelayerSetup(address relayer, bool approved);
```
Comment (LayneHaber) [inline packages/deployments/contracts/contracts/interfaces/IConnext.sol:139]: See above comments re: separate codepaths

File: packages/deployments/contracts/src/tasks/preflight.ts [hunk 5]
```diff
@@ -174,5 +178,15 @@ export default task("preflight", "Ensure correct setup for e2e demo with a speci
       } else {
         console.log("*** Sufficient relayer fees present!");
       }
+
+      // Add relayer
+      const relayer = _relayer ?? process.env.RELAYER_ADDRESS;
+      if (relayer && isAddress(relayer)) {
```
Comment (LayneHaber) [inline packages/deployments/contracts/src/tasks/preflight.ts:184]: should error if it is not an address

**Review by LayneHaber — CHANGES_REQUESTED** [review]

File: packages/deployments/contracts/contracts/Connext.sol (hunk 1 above)
Comment (liu-zhipeng) [reply packages/deployments/contracts/contracts/Connext.sol]: Trying to reduce contract bytecode size as much as I can. lol

Fixed.
[https://github.com/connext/nxtp/commit/ba57917bf911ee3be020d48634f3518ffe8a8750](url)

File: packages/deployments/contracts/src/tasks/preflight.ts (hunk 5 above)
Comment (liu-zhipeng) [reply packages/deployments/contracts/src/tasks/preflight.ts]: https://github.com/connext/nxtp/commit/6caf990bba55cce35f08d56ef5ecf378bcb872cf

**Pull request merged**